### ISSUE TITLE
Make Install Suddenly Very Expensive

### ISSUE THREAD

**Issue by bska** [OPM/opm-common#2340]
I'm suddenly seeing this in my local builds (release mode)
```
  --> cmake --build . --target all -- -k -j11
      Time = 214 [s]: Success
  --> cmake --build . --target install -- -j1
      Time = 139 [s]: Success
```
Until recently, the "make install" step was in the order of one second (or less), so something changed.  This **may** be related to the recent PR #2317, but I have not looked into that closely enough to be sure.

In any case, I'm now seeing the transcript below during "install" for opm-common which suggests that we're now generating and compiling all the `ParserKeywords/x.cpp` files as part of "make install".

---

```
[  5%] Built target genkw
[  5%] Updating version information
[  5%] Built target update-version
Scanning dependencies of target opmcommon
[  5%] Generating ParserKeywords/A.cpp, ParserKeywords/B.cpp, ParserKeywords/C.cpp, ParserKeywords/D.cpp, ParserKeywords/E.cpp, ParserKeywords/F.cpp, ParserKeywords/G.cpp, ParserKeywords/H.cpp, ParserKeywords/I.cpp, ParserKeywords/J.cpp, ParserKeywords/K.cpp, ParserKeywords/L.cpp, ParserKeywords/M.cpp, ParserKeywords/N.cpp, ParserKeywords/O.cpp, ParserKeywords/P.cpp, ParserKeywords/Q.cpp, ParserKeywords/R.cpp, ParserKeywords/S.cpp, ParserKeywords/T.cpp, ParserKeywords/U.cpp, ParserKeywords/V.cpp, ParserKeywords/W.cpp, ParserKeywords/X.cpp, ParserKeywords/Y.cpp, ParserKeywords/Z.cpp, TestKeywords.cpp, ParserInit.cpp, python/cxx/builtin_pybind11.cpp
[  5%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/Z.cpp.o
[  5%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/Y.cpp.o
[  6%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/X.cpp.o
[  6%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/W.cpp.o
[  6%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/V.cpp.o
[  6%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/U.cpp.o
[  6%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/T.cpp.o
[  6%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/S.cpp.o
[  6%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/R.cpp.o
[  7%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/Q.cpp.o
[  7%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/P.cpp.o
[  7%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/O.cpp.o
[  7%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/N.cpp.o
[  7%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/M.cpp.o
[  7%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/L.cpp.o
[  7%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/K.cpp.o
[  8%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/J.cpp.o
[  8%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/I.cpp.o
[  8%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/H.cpp.o
[  8%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/G.cpp.o
[  8%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/F.cpp.o
[  8%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/E.cpp.o
[  8%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/D.cpp.o
[  9%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/C.cpp.o
[  9%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/B.cpp.o
[  9%] Building CXX object CMakeFiles/opmcommon.dir/ParserKeywords/A.cpp.o
[  9%] Building CXX object CMakeFiles/opmcommon.dir/ParserInit.cpp.o
[  9%] Linking CXX static library lib/libopmcommon.a
```

**Comment by GitPaean** [comment]
> In any case, I'm now seeing the transcript below during "install" for opm-common which suggests that we're now generating and compiling all the ParserKeywords/x.cpp

I saw this today whenever I run the executable flow in my IDE even without any changes made.  I mean, without changing of source code, why it always need to recompile `ParserKeywords/x.cpp`?

**Comment by joakim-hove** [comment]
I have looked into this - it works as it should on my machine:
1. `make install` is fast
2. `ParserKeywords/x.cpp` is not recompiled.

My guess is that #2317 introduced cmake changes to the extent that something more powerful than `make clean` is required to get a consistent `cmake` project - just try wiping the build directory?

**Comment by bska** [comment]
> try wiping the build directory

No change.  The transcript below is from a completely new build directory and I still see the "generate & compile" stages in my "install" logs.  For the record, this build directory does **not** enable Python bindings.

---

```
--> new
  opt/seq/dune-2.7.1/opm-common
  --> cmake -DGCC_WARN_SHADOW=ON /home/bska/work/opm/src/opm/opm-common
      Time = 2.76 [s]
  --> cmake --build . --target all -- -k -j11
      Time = 292 [s]: Success
  --> cmake --build . --target install -- -j1
      Time = 138 [s]: Success

  Time total = 433 [s]
```

**Comment by joakim-hove** [comment]
OK - I reproduced the problems when Python was disabled. Fix here: https://github.com/OPM/opm-common/pull/2342

**Comment by bska** [comment]
> Fix here: #2342

Confirmed.  With that patch I'm back to the "make install" performance that I was used to earlier.

```
--> new
  opt/seq/dune-2.7.1/opm-common
  --> cmake -DGCC_WARN_SHADOW=ON /home/bska/work/opm/src/opm.master//opm-common
      Time = 2.78 [s]
  --> cmake --build . --target all -- -k -j11
      Time = 270 [s]: Success
  --> cmake --build . --target install -- -j1
      Time = 3.26 [s]: Success
```
